server/controllers/alertsController.ts

```diff
@@ -34,8 +36,14 @@ export default class AlertsController {
     // Get prisoner data for banner and for use in alerts generation
     const prisonerData = await this.prisonerSearchService.getPrisonerDetails(req.params.prisonerNumber)
 
+    // Set role based permissions
+    const { authorities: roles = [] } = jwtDecode(res.locals.user.token) as { authorities?: string[] }
+    const canUpdateAlert =
+      roles.some(role => role === Role.UpdateAlert) &&
+      (res.locals.user.activeCaseLoadId === prisonerData.prisonId || ['OUT', 'TRN'].includes(prisonerData.prisonId))
```

Comment:
Hmmm - The API when dealing with these two prison (/agency) IDs calls out a particular role (`INACTIVE_BOOKINGS`) - https://github.com/ministryofjustice/prison-api/blob/main/src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java#L541-L543

Is that something we might want to care about here? I'm not sure

Reply:
Good question - I don't know in this case. The Case Notes roles logic does look for that role and those IDs, so guessing it's fine here as-is, but we'll see if anything comes out of testing with more realistic cases/roles.